### ISSUE TITLE
MongoEngine + mongita

### ISSUE THREAD

**Issue by iraytrace** [scottrogowski/mongita#23]
While it is possible to use MongoEngine 0.22.1 with mongita-1.1.0 MongitaClientMemory:
```
import pymongo
import mongita
#  This works
pymongo.MongoClient = mongita.MongitaClientMemory
import mongoengine
mongoengine.connect(host='c:/temp/mongita')
```

It does not work with MongitaClientDisk:
```
import pymongo
import mongita
# This fails
pymongo.MongoClient = mongita.MongitaClientDisk
import mongoengine
mongoengine.connect(host='c:/temp/mongita')
```

The MongitaClientDisk constructor always fails.

Not clear if this is just a version mis-match with mongoengine or not.
The error also affects the unit tests for mongita.

The problem seems to be in mongita_client.py in the MongitaClientDisk constructor where it 
invokes:

```
disk_engine.DiskEngine.create(host)
```

This is because according to:
   https://pymongo.readthedocs.io/en/stable/api/pymongo/mongo_client.html
the _host_ parameter to the MongoClient() method is a list, not a string.  Since the
DiskEngine.create() factory method is expecting a string, it reports the error:
"unhashable type: list"

The simple fix is to test in the MongoClientDisk constructor in mongita_client.py
to see if the parameter is a list and if so pluck the first element.  Since the 
default value for mongoengine is 'localhost' this is worth checking for and replacing
with DEFAULT_STORAGE_DIR.

While we are at it, a check for the existence of the parent directory where the
database is to be located is worthwhile.  This leaves us with:

```
    def __init__(self, host=DEFAULT_STORAGE_DIR, **kwargs):
        host = host or DEFAULT_STORAGE_DIR
        if host == 'localhost':
            host = DEFAULT_STORAGE_DIR
 
         if os.path.exists(os.path.dirname(host)):
            raise NotADirectoryError(os.path.dirname(host))
 
        self.engine = disk_engine.DiskEngine.create(host)
```

I'd be happy to generate a pull request, but I'd like to know that the pre-existing unit tests work on some system before
requesting a pull for a fix that may be out of date.

**Comment by scottrogowski** [comment]
Hey, sorry to have dropped the ball on this. My laptop is broken and I’m traveling so I haven’t been able to validate the unit tests on my end locally which ultimately is a small point. If you’re still up for it and it’s a matter of fixing the unit tests, I would be happy to review a PR. I return to my country in 4 days so could review shortly afterwards

**Comment by iraytrace** [comment]
Understood.  At this point I'm now booked until December, but I'd take a look at it then.

**Comment by scottrogowski** [comment]
I finally got around to this and opened a PR to address the core issue you found https://github.com/scottrogowski/mongita/pull/24. Thank you again for the detailed report. I also took a stab at addressing the windows unit tests on a borrowed computer and it's looking like there might be deeper problems. In short, one trick for squeaking out more speed that Mongita employs is leaving file handles open. In the unit tests, these files can then be removed even when the handle is open. In windows, that is forbidden. 

The simple answer would be actually closing these file handles before deleting the temporary directories in the unit tests. I have a suspicion though that Mongita deletes files without closing handles in normal operation. If that were true, it would cause issues in a Windows environment.

For now, I'm going to slap a mac/linux only sticker on the readme since I can't verify how well this works on windows. In the next few weeks / months, I'll be attempting to discard the custom database and actually use SQLite as the database engine. If that works, I think it will solve the windows file handle issue at the same time.
